src/components/TimelineList/TimelineList.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import './TimelineList.scss'
+import { TimelineInterval } from '../TimelineInterval/TimelineInterval'
+import { calculateTimeIntervals } from '../../utils/calculateTimeIntervals'
+
+export const TimelineList = ({
+  intervalListWidthInPixels,
+  timeIntervals,
+  timelineListRef,
+  timelinePosition,
+  timelineWrapperRef,
+  zoomLevel,
+  onTimelineMouseDown
+}) => {
+  if (!timelineWrapperRef.current) return null
+
+  return (
+    <div
+      ref={timelineListRef}
+      className="timeline__list"
+      style={{
+        width: `${intervalListWidthInPixels}px`,
+        transform: `translateX(${timelinePosition.left}px)`
+      }}
+      onMouseDown={onTimelineMouseDown}
+      role="button"
+      tabIndex="0"
+    >
+      {
+        timeIntervals && timeIntervals.map((interval, intervalIndex) => {
+          const startTime = interval
+          let endTime
+
+          if (timeIntervals[intervalIndex + 1] != null) {
+            endTime = timeIntervals[intervalIndex + 1]
+          } else {
+            const lastInterval = timeIntervals[timeIntervals.length - 1]
```

Comment:
Create a second util on top of `getIntervalsDurations` where one returns the duration and one returns the start and end time.